### PR TITLE
gromacs: add 2021.7, 2022.5, 2023, and some requirements

### PR DESCRIPTION
**Pull request by zzzoom**
- Add version 2022.5 and 2023
- 2022 deprecated support for Intel classic compilers (https://manual.gromacs.org/2022-current/release-notes/2022/major/portability.html#intel-classic-compiler-icc-icpc-no-longer-supported)
- 2023 requires GCC 9 and oneMKL 2021.3 (https://manual.gromacs.org/current/release-notes/2023/major/portability.html#increase-of-required-versions)

### CODE REVIEW

**Comment by al42and** [comment]
LGTM.

A few suggestions:

- 2021.7 has also been released recently: `version("2021.7", sha256="4db7bbbfe5424de48373686ec0e8c5bfa7175d5cd74290ef1c1e840e6df67f06")`
- I'm not a maintainer here, but previously, explicitly listing all the conflicts was not a practice here. If that becomes the norm, the [known issues](https://manual.gromacs.org/2023/user-guide/known-issues.html) should also be accounted for, I think.

**Review by tldahlgren — DISMISSED** [review]
Confirmed new sha256, rest LGTM

**Comment by tldahlgren** [comment]
Deferring merge decision to give @marvinbernhardt and or @junghans time to comment.

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -140,6 +142,21 @@ class Gromacs(CMakePackage):
         when="+cp2k",
         msg="GROMACS and CP2K should use the same blas, please disable bundled blas",
     )
+    conflicts(
+        "%intel",
+        when="@2022:",
+        msg="GROMACS %intel support was removed in version 2022",
```
Comment (junghans) [inline var/spack/repos/builtin/packages/gromacs/package.py:149]: @mabraham is that right?

**Comment by zzzoom** [comment]
>     * 2021.7 has also been released recently: `version("2021.7", sha256="4db7bbbfe5424de48373686ec0e8c5bfa7175d5cd74290ef1c1e840e6df67f06")`

Done

>     * I'm not a maintainer here, but previously, explicitly listing all the conflicts was not a practice here. If that becomes the norm, the [known issues](https://manual.gromacs.org/2023/user-guide/known-issues.html) should also be accounted for, I think.

AFAIK Spack has no way to describe some of the problematic configurations listed there (Ubuntu 22.04's flavor of GCC 11, oneMKL 2023.0 being used as the FFT instead of BLAS/LAPACK). CUDA 11.3.0 could be tagged as a conflict but I couldn't find if 11.3.1 which is newer than the bug report fixed the issue.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (marvinbernhardt) [reply var/spack/repos/builtin/packages/gromacs/package.py:149]: This seems right from comparing [2021 docu](https://manual.gromacs.org/2021/install-guide/index.html) and [2022 docu](https://manual.gromacs.org/2022/install-guide/index.html).

**Comment by marvinbernhardt** [comment]
I think it is fine to have some basic checks for things like compiler version. The more detailed issues (CUDA, Ubuntu gcc, ..) do not have to be included in spack in my opinion.

**Review by marvinbernhardt — APPROVED** [review]
2021.7 checksum confirmed

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (mabraham) [reply var/spack/repos/builtin/packages/gromacs/package.py:149]: Yes, the Intel classic compiler is no longer supported by GROMACS. The LLVM-based ICPX/DPCPP compiler is supported by GROMACS.

**Review by mabraham — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Confirmed the three affected version's sha256.

**Comment by tldahlgren** [comment]
@spackbot run pipeline

**Pull request merged**